### ISSUE TITLE
`--no-ignore` not work with `--stdin-filename`

### ISSUE THREAD

**Issue by muzuiget** [eslint/eslint#12745]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
$ cd ~
$ touch .hello.js
$ eslint .hello.js # ignored, expected
$ eslint --no-ignore .hello.js # not ignored, expected
$ cat .hello.js | eslint --no-ignore --stdin --stdin-filename .hello.js # ignored, not expected
```
I am using Vim plugin [ale](https://github.com/dense-analysis/ale), this plugin always call `eslint` with  `--stdin-filename`, even I append `--no-ignore` to ale option, eslint still ignore the file, so all the files in my $HOME always be ignored when it is opened in Vim.

**What did you expect to happen?**

File shouln't be ignored with below command:

```bash
$ cat .hello.js | eslint --no-ignore --stdin --stdin-filename .hello.js 
```

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by kaicataldo** [comment]
I have confirmed this behavior and agree that this looks like a bug.

**Comment by karthikkp** [comment]
@kaicataldo @muzuiget I have taken up this issue and put up a PR that fixes this issue.

**Comment by kaicataldo** [comment]
Assigning myself so that it doesn't get auto closed again.
